developer_tools/cargo/cargo.py

```diff
@@ -0,0 +1,504 @@
+    output of `cargo outdated`
+    the keys are the string representations of dependencies
+    the values are 4-tuples containing
+    1) the string represenation of the dependency's version (i.e. from the
+    "Project" column of the output of `cargo outdated`,
+    2) the string representation of the dependency the dependency is pulled in by
+    or None if the dependency is not pulled in by any dependency
+    3) the string representation of the dependency's platform information (i.e.
+    from the "Platform" column of the ouput of `cargo outdated`
+    4) a bool indicating whether or not the dependency should be "included", with
+    respect to the platform information
+    :type cargo_outdated_dict: dict
+    :param koji_repo_dict: a dictionary containing information from the koji repo webpage
+    the keys are the string representations of dependencies
+    the values are the string representations of versions of dependencies
+    :type koji_repo_dict: dict
+    """
+
+    outdated = []
+    not_outdated = []
+    not_found = []
+    not_included = []
+    table_data = []
+
+    table_data.append(
+        ["Crate", "Outdated?", "Current", "Update To", "Include?", "Platform"]
+    )
+    table_data.append(
+        ["-----", "---------", "-------", "---------", "--------", "--------"]
+    )
+
+    for key in cargo_outdated_dict:
+
+        version = cargo_outdated_dict[key][0]
+        platform = cargo_outdated_dict[key][2]
+        include = cargo_outdated_dict[key][3]
+
+        if key in koji_repo_dict.keys():
+            if koji_repo_dict[key] != version:
+                table_data.append(
+                    [
+                        key,
+                        "Outdated",
+                        version,
+                        koji_repo_dict[key],
+                        str(include),
+                        platform,
+                    ]
+                )
+                if include:
+                    outdated.append(key)
+                else:
+                    not_included.append(key)
+
+            else:
+                table_data.append(
+                    [
+                        key,
+                        "Not Outdated",
+                        version,
+                        koji_repo_dict[key],
+                        str(include),
+                        platform,
+                    ]
+                )
+                if include:
+                    not_outdated.append(key)
+                else:
+                    not_included.append(key)
+
+        else:
+            table_data.append(
+                [key, "Not Found", version, "---", str(include), platform]
+            )
+            if include:
+                not_found.append(key)
+            else:
+                not_included.append(key)
+
+    print("\n\nRESULTS")
+
+    print(
+        "\nThe following crates that were outputted by 'cargo outdated' are outdated"
+        + " with respect to the koji repo:"
```

Comment:
There should be a way to get rid of this '+'. Try just removing it and leaving the rest the same and see if the program still runs.